src/node/internal/events.ts

```diff
@@ -745,7 +760,9 @@ export function listenerCount(emitter: any, type: string | symbol) {
 }
 
 EventEmitter.prototype.eventNames = function eventNames() {
-  return this._eventsCount > 0 ? Reflect.ownKeys(this._events) : [];
+  return this._eventsCount > 0 && this._events !== undefined
+    ? Reflect.ownKeys(this._events)
+    : [];
```

Comment:
Nit: I believe you could simplify this a bit with...

```suggestion
  return this._eventsCount > 0 
    ? Reflect.ownKeys(this._events || {})
    : [];
```

Reply:
Agreed